### PR TITLE
client side file

### PR DESCRIPTION
**Pull request by Mberenice**
worked on client.java, CreateAccountControl.java, GameControl.java, GameLobbyControl.java, GameManagerControl.java, GameSessionControl.java, loginControl.java, Startupcontrol.java

### CODE REVIEW

File: UNO/ClientCommunication/Client.java [hunk 1]
```diff
@@ -2,4 +2,46 @@
 
 public class Client {
 
+<<<<<<< Updated upstream
```
Comment (madzmadon) [inline UNO/ClientCommunication/Client.java]: Remove upstream comments

**Review by madzmadon — CHANGES_REQUESTED** [review]

**Review by madzmadon — CHANGES_REQUESTED** [review]
Could  you add comments to describe what is happening in different pieces of the code?

**Review by LPitt1 — COMMENTED** [review]
Possible changes:
- The controllers are used as event handlers. Sending them through the client/server connection seems to undermine this purpose. To solve this, we could use the LoginData and CreateAccountData classes; however, they would need to be made serialized.
- Passing the UI elements instead of just their strings would help if the controller classes needed more control over the UI.
- The client class needs to be extended from the abstract client class from OCSF and; therefore, the required method would need to be overridden.

File: UNO/ClientCommunication/GameSessionControl.java [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+    public GameSessionControl(Client client, String lobbyId) {
+        this.client = client;
+        this.lobbyId = lobbyId;
+    }
+
+    
+    public void updateGameState() {
+        Object gameStateUpdate = client.receiveGameState();  // Simulating receiving a game state
+        System.out.println("Game state updated for lobby " + lobbyId + ": " + gameStateUpdate);
+    }
```
Comment (madzmadon) [inline UNO/ClientCommunication/GameSessionControl.java:77]: **Recommended to add**

 Fields
    - game session data
    - List of players in the game
    - Index of the current player
 public void addPlayer(ClientHandler player) 

 public void removePlayer(ClientHandler player) 
    
public ClientHandler getCurrentPlayer() 

public void nextTurn() 

public boolean isValidMove(Card card) 
 -Check if the played card is a valid move based on the game rules
 -Example: Check if the card matches the color or value of the top card in the discard pile

public void playCard(ClientHandler player, Card card) 
 - Remove the card from the player's hand
 - Add the card to the discard pile
 -if card type equals "Draw Two, Draw Four", next player draws cards
-Add more conditions for other special card effects

public Card drawCard() 

public void callUno(ClientHandler player) 
- Handle the player calling UNO
-Example: Check if the player has only one card left and apply any penalties if they didn't call UNO in time

public boolean isGameOver() 
-Check if the game is over
-Example: Check if any player has an empty hand

public void broadcastMessage(String message) 
-send players valid messages from server for when they play a bad card, the server will send you this message, you just have to display it

public GameSessionData getGameSessionData() 

private ClientHandler getNextPlayer()

**Review by madzmadon — CHANGES_REQUESTED** [review]

File: UNO/ClientCommunication/GameControl.java [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
+        Object gameState = client.receiveGameState();
+        handleGameStateUpdate(gameState);
+    }
+
+    
+    private void handleGameStateUpdate(Object gameState) {
+        
+        System.out.println("Received game state update: " + gameState);
+    }
 }
```
Comment (madzmadon) [inline UNO/ClientCommunication/GameControl.java:43]: // add gameId to Fields
private int gameId;

add this.gameId = gameId; to constructor

// Method to get the game ID
public int getGameId() {... }

**Review by madzmadon — CHANGES_REQUESTED** [review]

**Pull request merged**